src/test/java/com/google/cloud/spanner/pgadapter/python/PythonTransactionTests.java

```diff
@@ -0,0 +1,1095 @@
+    Assert.assertEquals(1, Collections.frequency(sql, sql3));
+    Assert.assertEquals(1, Collections.frequency(sql, sql4));
+  }
+
+  // Isolation Levels
+  // ISOLATION_LEVEL_AUTOCOMMIT -> 0
+  // ISOLATION_LEVEL_READ_COMMITTED -> 1
+  // ISOLATION_LEVEL_REPEATABLE_READ -> 2
+  // ISOLATION_LEVEL_SERIALIZABLE -> 3
+  // ISOLATION_LEVEL_READ_UNCOMMITTED -> 4
+  @Test
+  public void testIsolationLevelErrorInTransactions() throws IOException, InterruptedException {
+    // tests isolation_level settings using connection.isolation_level variable
+    List<String> statements = new ArrayList<>();
+    String sql1 = "Select * from some_table";
+    String sql2 = "insert into some_table(col1, col2) values(value1, value2)";
+    String sql3 = "Select * from some_table3";
+    String sql4 = "insert into some_table3(col1, col2) values(value1, value2)";
+
+    // query will be executed as expected
+    statements.add("query");
+    statements.add(sql1);
+
+    // update will be executed as expected
+    statements.add("update");
+    statements.add(sql2);
+
+    // update will be executed as expected
+    statements.add("update");
+    statements.add(sql4);
+
+    // 1 commit request will be sent
+    statements.add("transaction");
+    statements.add("commit");
+
+    statements.add("transaction");
+    statements.add("set isolation_level 1");
+
+    // query won't be executed because the previous setting would've thrown error
+    statements.add("query");
+    statements.add(sql3);
+
+    mockSpanner.putStatementResult(
+        StatementResult.query(Statement.of(sql1), createResultSet(1, "abcd")));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(sql2), 1));
+    mockSpanner.putStatementResult(
+        StatementResult.query(Statement.of(sql3), createResultSet(2, "pqrs")));
+    mockSpanner.putStatementResult(StatementResult.update(Statement.of(sql4), 2));
+
+    String expectedOutput =
+        "(1, 'abcd')\n"
+            + "1\n"
+            + "2\n"
+            + "INVALID_ARGUMENT: Unknown value for TRANSACTION: ISOLATION LEVEL READ COMMITTED\n";
+    String actualOutput = executeTransactions(pgServer.getLocalPort(), statements);
+    Assert.assertEquals(expectedOutput, actualOutput);
+
+    Assert.assertEquals(1, mockSpanner.countRequestsOfType(CommitRequest.class));
+    Assert.assertEquals(0, mockSpanner.countRequestsOfType(RollbackRequest.class));
+    Assert.assertEquals(3, mockSpanner.countRequestsOfType(ExecuteSqlRequest.class));
+
+    List<String> sql = new ArrayList<>();
+
+    for (ExecuteSqlRequest request : mockSpanner.getRequestsOfType(ExecuteSqlRequest.class)) {
+      sql.add(request.getSql());
+    }
+
+    Assert.assertEquals(1, Collections.frequency(sql, sql1));
+    Assert.assertEquals(1, Collections.frequency(sql, sql2));
+    Assert.assertEquals(0, Collections.frequency(sql, sql3));
+    Assert.assertEquals(1, Collections.frequency(sql, sql4));
+
+    mockSpanner.clearRequests();
```

Comment:
Can we split this test into three separate test cases:
1. A test for the isolation levels that we don't support (so 2 through 4) and try each of them, possibly just in a loop.
2. A test for the isolation level(s) that we do support (currently only 1)
3. A separate test for setting it to level 0. I wonder exactly what SQL statement the driver is generating if you try to set autocommit by setting the isolation level to 0.

Reply:
Done. 
Actually, when setting it to 0, it will behave just like setting `autocommit = True`. So, no particular statement will be sent.